Make MockSpanBuilder public so it can be used in different packages

pavolloffay opened the pull request:
`MockTracer.SpanBuilder` has package level visibility. It cannot be used in different packages (`io.opentracing.mock`) . It makes it hard to test instrumentation libraries which are usually in different packages.

Q:
`MockContext` should be probably also public. It is needed when implementing custom `MockTracer.Propagator` (which a paramter in `MockTracer` public constructor).

@yurishkuro @bensigelman could you please review?

yurishkuro commented:
What is the use case where you need to use the concrete type `MockTracer.SpanBuilder`?

I agree that MockContext should be public.

pavolloffay commented:
PR updated.

> What is the use case where you need to use the concrete type MockTracer.SpanBuilder?

I changed the code, `MockTracer#buildSpan()` should return `Tracer.SpanBuilder` and not concrete impl, which has package visibility.

> I agree that MockContext should be public.

I have changed `MockContext` constructor to be public. This enables implementing custom `MockTracer.Propagator` in different packages.

yurishkuro commented on opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java:
```diff
@@ -100,7 +100,7 @@ protected void onSpanFinished(MockSpan mockSpan) {
 
     @Override
-    public SpanBuilder buildSpan(String operationName) {
+    public Tracer.SpanBuilder buildSpan(String operationName) {
```
hm, strictly speaking this _reduces_ the visibility since you're returning an interface even when you invoke the method directly via mock tracer instance. Was that your intention?

pavolloffay replied:
`mockTracer.buildSpan("parent").start()` fails in packages different what `io.opentracing.mock` because mock `SpanBuilder` has package visibility.

There are two options: 
* change visibility of mock `SpanBuilder` to public
* `MockTracer#buildSpan` return `Tracer.SpanBuilder`
